### PR TITLE
Form created using react

### PR DESCRIPTION
**Pull request by linhapeter**

### CODE REVIEW

File: src/Api.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import { useState } from "react";
+import axios from "axios";
+
+const Api = () => {
+  const [data, setData] = useState(null);
```
Comment (volkov-a-dev) [inline src/Api.js]: @linhapeter There is no need to useState in API methods. This needs to be removed. This logic should be in the component container

File: src/Api.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  const [error, setError] = useState(null);
```
Comment (volkov-a-dev) [inline src/Api.js]: needs to be removed

File: src/Api.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  const [loading, setLoading] = useState(null);
```
Comment (volkov-a-dev) [inline src/Api.js]: needs to be removed

File: src/Api.js [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+  const fetchData = async (url, method = "get", body = null) => {
```
Comment (volkov-a-dev) [inline src/Api.js]: You need to remove useStates from the fetchData function. Also, for readability, it is better to transform the query function into this form.

```
export const fetchData = (params) =>
  axios(params)
    .then(({ data }) => {
      return data;
    })
    .catch((error) => {
      console.log("error", error);
      throw error;
    });
```
By default, axios uses the get method. So there's no point in adding get. [Documentation](https://axios-http.com/docs/example)

File: src/Form.js [hunk 5]
```diff
@@ -0,0 +1,59 @@
+import React, { useState } from "react";
+// import ResponseComponent from './ResponseComponent';
+import Api from "./Api.js";
+const Form = () => {
+  const [name, setName] = useState("");
+  const [password, setPassword] = useState("");
+
+  const { data: response, error, loading, fetchData } = Api();
```
Comment (volkov-a-dev) [inline src/Form.js]: Not quite the right way to get data. Please read the [article](https://www.codingthesmartway.com/how-to-fetch-api-data-with-react/)

File: src/Form.js [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+  const submitForm = async (e) => {
```
Comment (volkov-a-dev) [inline src/Form.js]: This logic should not be in the component.
This request function(submitForm) must be on the API file.

File: src/Form.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+    e.preventDefault();
+
+    fetchData("https://httpbin.org/post", "post", { name, password });
+  };
+  return (
+    <div>
+      <form onSubmit={submitForm}>
```
Comment (volkov-a-dev) [inline src/Form.js]: It's better to move this into a separate component Form.

File: src/Form.js [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        <button type="submit" disabled={loading}>
+          Submit
+        </button>
+      </form>
+      {loading ? (
+        "Loading..."
+      ) : (
+        <div>
+          {error && <div>Error: {error.message}</div>}
+          {response && (
```
Comment (volkov-a-dev) [inline src/Form.js]: It's better to move this into a separate component.

**Comment by volkov-a-dev** [comment]
I noticed that all files have JS extensions. However, it is better to use the JSX extension for react components.
[What is the difference between JS and JSX?](https://www.linkedin.com/pulse/what-difference-between-js-jsx-skill-gain/)

**Comment by linhapeter** [comment]
I've changed extensions. Is everything else ok?

**Review by volkov-a-dev — CHANGES_REQUESTED** [review]

File: src/Form.jsx [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+  const submitForm = async (e) => {
+    e.preventDefault();
+
+    setLoading(true);
+    const params = {
+      url: "https://httpbin.org/post",
+      method: "post",
+      data: { name, password },
+    };
+    setResponse(await fetchData(params));
```
Comment (volkov-a-dev) [inline src/Form.jsx]: You shouldn't do this, it's bad practice.
What happens if the server returns a 404 or 500 error?

It's better to put this in a separate variable and pass it to the setResponse(useState)

File: src/Form.jsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+  const [name, setName] = useState("");
+  const [password, setPassword] = useState("");
+  const [response, setResponse] = useState(null);
+  const [loading, setLoading] = useState(null);
+
+  const submitForm = async (e) => {
+    e.preventDefault();
+
+    setLoading(true);
+    const params = {
```
Comment (volkov-a-dev) [inline src/Form.jsx:14]: It’s better to pass the search parameters and put everything else in the API
`const params = { name, password }`

File: src/Api.jsx [hunk 11]
```diff
@@ -0,0 +1,11 @@
+import axios from "axios";
+
+export const fetchData = async (params) =>
```
Comment (volkov-a-dev) [inline src/Api.jsx:5]: Why not write it like this? Is it more convenient to maintain code in such an implementation? Is it convenient to expand it?
**Example**

```
const USER_LOGIN_API = "https://httpbin.org/post";
export const setUserLogin = async (params) =>
  axios
    .post(USER_LOGIN_API, params)
    .then(({ data }) => data)
    .catch((error) => {
      throw error;
    });
```

File: src/Form.jsx [hunk 12]
```diff
@@ -0,0 +1,51 @@
...
+            onChange={(e) => setName(e.target.value)}
+          />
+        </label>
+        <br />
+        <label>
+          Password:
+          <input
+            type="password"
+            value={password}
+            onChange={(e) => setPassword(e.target.value)}
```
Comment (volkov-a-dev) [inline src/Form.jsx]: Please read this article
https://www.linkedin.com/pulse/handling-events-react-best-practices-common-mistakes-faizan/

File: src/Form.jsx
Comment (volkov-a-dev) [inline src/Form.jsx:1]: It's better to separate business logic and rendering(components, container)  Please read this [article](https://www.freecodecamp.org/news/separation-of-concerns-react-container-and-presentational-components/)

File: src/ResponseContent.jsx [hunk 13]
```diff
@@ -0,0 +1,16 @@
+import React from "react";
+const ResponseContent = ({ response }) => (
+  <div>
+    {response && (
+      <div>
+        <strong>Response:</strong>
+        <ul>
+          <li>Name: {response.json.name}</li>
```
Comment (volkov-a-dev) [inline src/ResponseContent.jsx:8]: What happens if the response saves this data? Please test `response = {}`

File: src/Form.jsx [hunk 14]
```diff
@@ -0,0 +1,51 @@
+import React, { useState } from "react";
+import { fetchData } from "./Api.jsx";
+import ResponseContent from "./ResponseContent.jsx";
+const Form = () => {
+  const [name, setName] = useState("");
+  const [password, setPassword] = useState("");
+  const [response, setResponse] = useState(null);
+  const [loading, setLoading] = useState(null);
```
Comment (volkov-a-dev) [inline src/Form.jsx]: The code must be self-documenting. [Naming conventions](https://www.w3schools.com/js/js_conventions.asp).
const [isLoading, setIsLoading] = useState(false);

**Review by volkov-a-dev — CHANGES_REQUESTED** [review]

File: src/Form.jsx [hunk 15]
```diff
@@ -0,0 +1,58 @@
...
+    setLoading(false);
+  };
+
+  const hanndleNameFieldChange = (e) => {
+    e.preventDefault();
+    setName(e.target.value);
+  };
+
+  const hanndlePasswordFieldChange = (e) => {
+    e.preventDefault();
```
Comment (volkov-a-dev) [inline src/Form.jsx]: Why is this here? `  e.preventDefault();`

File: src/Form.jsx [hunk 16]
```diff
@@ -0,0 +1,58 @@
...
+      name,
+      password,
+    };
+    const answer = await fetchData(params);
+    setResponse(answer);
+    setLoading(false);
+  };
+
+  const hanndleNameFieldChange = (e) => {
+    e.preventDefault();
```
Comment (volkov-a-dev) [inline src/Form.jsx]: Why is this here? `  e.preventDefault();`

**Review by volkov-a-dev — CHANGES_REQUESTED** [review]

File: src/Form.jsx (hunk 16 above)
Comment (linhapeter) [reply src/Form.jsx]: I've read that it's nice to have to prevent default behavior. But I guess I've misunderstood.

File: src/Form.jsx [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+    const params = {
+      name,
+      password,
+    };
+    const answer = await fetchData(params);
+    setResponse(answer);
+    setLoading(false);
+  };
+
+  const hanndleNameFieldChange = (e) => {
```
Comment (volkov-a-dev) [inline src/Form.jsx]: hanndleNameFieldChange = > 
Many developers find incorrectly named variables frustrating, and this can lead to unnecessary negativity. To avoid this issue, it's advisable to consider using text verification extensions. If you use VSCode, I would recommend installing the extension to help maintain clean and consistent code.
[Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker)

File: src/Form.jsx [hunk 18]
```diff
@@ -0,0 +1,56 @@
...
+    setName(e.target.value);
+  };
+
+  const hanndlePasswordFieldChange = (e) => {
```
Comment (volkov-a-dev) [inline src/Form.jsx]: hanndleNameFieldChange = > 
Many developers find incorrectly named variables frustrating, and this can lead to unnecessary negativity. To avoid this issue, it's advisable to consider using text verification extensions. If you use VSCode, I would recommend installing the extension to help maintain clean and consistent code.
[Code Spell Checker](https://marketplace.visualstudio.com/items?itemName=streetsidesoftware.code-spell-checker)

File: src/Form.jsx [hunk 19]
```diff
@@ -0,0 +1,56 @@
...
+    setPassword(e.target.value);
+  };
+
+  return (
+    <div>
+      <form onSubmit={submitForm}>
+        <label>
+          Name:
+          <input type="text" value={name} onChange={hanndleNameFieldChange} />
```
Comment (volkov-a-dev) [inline src/Form.jsx]: The problem is described above

**Review by volkov-a-dev — CHANGES_REQUESTED** [review]
